### PR TITLE
Store non-sensitive UI state without protection

### LINKED ISSUE #4949: Protected storage configuration is lost between dashboard runs from a container
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The dashboard uses protected storage to save orientation/size of details views. This data is protected using data protection. Data protection keys are recreated each time the dashboard is run from the container in typical usage.

The end result is orientation/size of details views is lost between runs of the standalone dashboard. Theme selection doesn't have this problem because it is saved to a cookie.

Possible solutions

This data isn't sensitive and doesn't need to be encrypted. We can use a solution that doesn't do any encryption to fix this issue.

* Switch to using a cookie.
* Continue to use localStorage but don't protect saved data. I think it's important we **don't** completely replace `ProtectedLocalStorage` in DI with an instance that doesn't encrypt. It might be used internally by Blazor or other libraries to protect data that is actually sensitive. Perhaps have a keyed registration of `ProtectedLocalStorage` that isn't protected by specifying `IDataProtectionProvider` option that doesn't do anything.

### Expected Behavior

Orientation/size of details views is remembered.

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version info

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JamesNK**
## Description

Some UI state is stored in protected browser storage: details orientation and size.

Unfortunately protected storage becomes unusable if the dashboard's data protection keys change. This can happen if the dashboard is launched from a docker container (each run has new keys). I've also seen it when updating Aspire version - perhaps the dashboard being located in a different place or a different file name causes new keys? IDK.

The fix is to store long term (localStorage) non-sensitive state without protected. Values will be remembered between runs, and there won't be errors silently thrown in the background when attempting to unencrpyt and failing.

Short term storage (sessionStorage) that is used to store potentially sensitive page data still only allows storage with protection.

I'll double check this is safe with Blazor and security folks.

Fixes https://github.com/dotnet/aspire/issues/4949

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] Yes
    - If yes, have you done a threat model and had a security review?
      - [x] Yes
      - [ ] No
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5434)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs:295]: I changed the keys to prevent the dashboard from using existing encrypted values. This means existing state is lost when updating to the next Aspire version, but it appears to be lost today anyway when protection keys change.

Going forward we can rely on these values being available long term.

Comment (drewnoakes) [reply src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs:295]: While we're at it, can we give this a better name? There might be multiple splitters in the dashboard. Let's make this more specific.

File: src/Aspire.Dashboard/Model/BrowserStorage/LocalBrowserStorage.cs
Comment (drewnoakes) [inline src/Aspire.Dashboard/Model/BrowserStorage/LocalBrowserStorage.cs:45]: These two methods return `ValueTask`. I'm not sure how often these would return completed tasks, but perhaps we also use value tasks on the new methods on `ILocalStorage`.

File: src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs:295]: viewKey is there to support multiple splitters. It is what lets resources/structured logs/trace detail to have splitters with different settings.

**Review by drewnoakes — APPROVED** [review]
Approved, assuming the security review is completed before merge.

File: src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs
Comment (drewnoakes) [reply src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs:295]: Ok, makes sense.

**Comment by JamesNK** [comment]
> Approved, assuming the security review is completed before merge.

Done over email

**Pull request merged**
